app/main.py

```diff
@@ -1,6 +1,19 @@
-from typing import Callable
+from typing import Callable, Any
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_storage = {}
+
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> Any:
+        cache_key = (args, tuple(sorted(kwargs.items())))
```

Comment:
The cache key is created using `tuple(sorted(kwargs.items()))`, which assumes that all keyword arguments are hashable and that their order does not affect the result. If any of the keyword arguments are not hashable (e.g., lists or dictionaries), this will raise an error. Consider ensuring that all keyword arguments are hashable or converting them to a hashable type.